autogpt/memory/vector/providers/no_memory.py

```diff
@@ -0,0 +1,36 @@
+"""A class that does not store any data. This is the default memory provider."""
+from __future__ import annotations
+
+from typing import Iterator, Optional
+
+from autogpt.config.config import Config
+
+from .. import MemoryItem
+from .base import VectorMemoryProvider
+
+
+class NoMemory(VectorMemoryProvider):
+    """
+    A class that does not store any data. This is the default memory provider.
+    """
+
+    def __init__(self, config: Optional[Config] = None):
```

Comment:
I'm confused as to what the public interface of a `VectorMemoryProvider` is.  The base class only has some (implemented) methods for retrieving memory. These two implementations have _different_ sets of methods for interacting with the providers and saving data.  Their public interface should be identical and defined in the base class, otherwise they're not swappable in the rest of the codebase.

Reply:
Most of the interface comes from `MutableSet[MemoryItem]`, which defines and inherits abstract methods such as `add()`, `discard()`, `__iter__()`, `__len__()`, `__contains__()`.

Reply:
I see.